### PR TITLE
docs.managed resources: add more details to versioning policy

### PR DESCRIPTION
**Pull request by muvaf**
Signed-off-by: Muvaffak Onus <me@muvaf.com>

### Description of your changes

Provider developers ask when a resource needs to be bumped from alpha to beta. While the Kubernetes API Versioning doc is what we follow, more details seem to be helpful.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N/A

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/concepts/managed-resources.md [hunk 1]
```diff
@@ -439,9 +459,12 @@ including Velero.
...
-[api-versioning]: https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning
+[api-versioning]:
+    https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning
 [velero]: https://velero.io/
 [api-reference]: ../api-docs/overview.md
 [provider]: providers.md
 [issue-727]: https://github.com/crossplane/crossplane/issues/727
 [issue-1143]: https://github.com/crossplane/crossplane/issues/1143
+[managed-api-patterns]:
+    https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md
```
Comment (hasheddan) [inline docs/concepts/managed-resources.md:470]: I think this may break these links

File: docs/concepts/managed-resources.md [hunk 2]
```diff
@@ -241,6 +244,23 @@ conventions][api-versioning] for the CRDs that it deploys. In short, for
 instructions for manual migration will be provided when a new version of that
 CRD schema is released.
 
+In practice, we suggest the following guideline to provider developers:
```
Comment (hasheddan) [inline docs/concepts/managed-resources.md:247]: nitpick:
```suggestion
In practice, we suggest the following guidelines to provider developers:
```

File: docs/concepts/managed-resources.md [hunk 3]
```diff
@@ -241,6 +244,23 @@ conventions][api-versioning] for the CRDs that it deploys. In short, for
...
+* Every new kind has to be introduced as `v1alpha1` with no exception.
+* Breaking changes require a schema change, i.e. `v1alpha1` needs to become
```
Comment (hasheddan) [inline docs/concepts/managed-resources.md:249]: I think the schema change is implied here but we are saying that version bump is required
```suggestion
* Breaking changes require a version change, i.e. `v1alpha1` needs to become
```

File: docs/concepts/managed-resources.md (hunk 1 above)
Comment (muvaf) [reply docs/concepts/managed-resources.md:470]: Just checked that it doesn't

Comment (hasheddan) [reply docs/concepts/managed-resources.md:470]: Gotcha 👍🏻 Wasn't sure if it would break in site rendering.

**Review by hasheddan — APPROVED** [review]

**Review by jbw976 — APPROVED** [review]
Thank you for taking the time to write this up and add some clarity for versioning in our ecosystem 💪

**Pull request merged**